client/gutenberg/editor/init.js

```diff
@@ -24,6 +25,45 @@ const WPCOM_UNSUPPORTED_CORE_BLOCKS = [
 	'core/file', // see D19851 for more details.
 ];
 
+/* eslint-disable prefer-const */
+// this is indeed reasigned when unregistering blocks on applyGutenbergBlockAvailability
+let blockCache = {};
+
+const partitionBlocks = flow(
+	partialRight( map, ( value, key ) => ( { name: key, ...value } ) ),
+	partialRight( partition, 'available' ),
+	partialRight( map, blocks => map( blocks, ( { name } ) => name ) )
+);
```

Comment:
I'm admittedly not that used to this kind of FP (sequencing `flow` / `partialRight` in particular) so this took me a bit to wrap my head around :grimacing: